theano/sandbox/cuda/tests/test_dnn.py

```diff
@@ -582,6 +582,17 @@ def test_dnn_softmax_grad(self):
         if cuda.dnn.version() != (3000, 3000):
             utt.verify_grad(softmax_op, [x_val2], mode=mode_with_gpu)
 
+    def test_local_softmax_dnn_grad(self):
+        """
+        Check for optimization error when grad of summed
+        softmax is taken over tensor with fixed shape.
+        """
+        x = T.fvector('x')
+        xp = x.reshape((5, 5))
+        y = T.nnet.softmax(xp.flatten()).sum()
+        g = T.grad(y, x)
+        theano.function(inputs=[x], outputs=g, mode=self.mode)
+
```

Comment:
Can you add an assert that the optimized function contain the good op. Something like:

assert(any(n for n in f.maker.fgraph.toposort() if isinstance(n.op, GpuDnnSoftmaxGrad))

Otherwise, the PR seem good. thanks.